src/sentry/static/sentry/app/views/dashboardsV2/dashboard.tsx

```diff
@@ -476,11 +227,151 @@ const AddWidgetWrapper = styled('a')`
   justify-content: center;
 `;
 
-const WidgetGhost = styled('div')`
-  display: block;
-  position: absolute;
-  cursor: grabbing;
-  opacity: 0.8;
-  box-shadow: ${p => p.theme.dropShadowHeavy};
-  border-radius: ${p => p.theme.borderRadius};
-`;
+type SortableWidgetProps = {
+  widget: Widget;
+  dragId: string;
+  isEditing: boolean;
+  isDragging: boolean;
+  onDelete: () => void;
+  onEdit: () => void;
+};
+
+function SortableWidget(props: SortableWidgetProps) {
+  const {widget, dragId, isDragging, isEditing, onDelete, onEdit} = props;
+
+  const {
+    attributes,
+    listeners,
+    setNodeRef,
+    transform,
+    isDragging: currentWidgetDragging,
+  } = useSortable({
+    id: dragId,
+    transition: null,
+  });
+
+  useEffect(() => {
+    if (!currentWidgetDragging) {
+      return undefined;
+    }
+
+    document.body.style.cursor = 'grabbing';
+
+    return function cleanup() {
+      document.body.style.cursor = '';
+    };
+  }, [currentWidgetDragging]);
+
+  const initialStyles = {
+    x: 0,
+    y: 0,
+    scaleX: 1,
+    scaleY: 1,
+    boxShadow: 'none',
+    zIndex: 0,
+  };
+
+  return (
+    <WidgetWrapper
+      ref={setNodeRef}
+      displayType={widget.displayType}
+      layoutId={dragId}
+      animate={
+        transform
+          ? {
+              x: transform.x,
+              y: transform.y,
+              scaleX: transform?.scaleX && transform.scaleX <= 1 ? transform.scaleX : 1,
+              scaleY: transform?.scaleY && transform.scaleY <= 1 ? transform.scaleY : 1,
+              zIndex: currentWidgetDragging ? theme.zIndex.modal : 0,
+              boxShadow: currentWidgetDragging
+                ? '0 0 0 1px rgba(63, 63, 68, 0.05), 0px 15px 15px 0 rgba(34, 33, 81, 0.25)'
```

Comment:
Should this be defined in theme so we can use it on other draggable things?

Reply:
I took this from dndkit's example.

I replaced it with `theme.dropShadowHeavy` instead.